networkx/algorithms/tree/mst.py

```diff
@@ -0,0 +1,313 @@
+
+    >>> G = nx.cycle_graph(4)
+    >>> G.add_edge(0, 3, weight=2)
+    >>> mst = tree.kruskal_mst_edges(G, minimum=True, data=False)
+    >>> sorted(mst)
+    [(0, 1), (1, 2), (2, 3)]
+
+    Find a maximum spanning tree.
+
+    >>> G = nx.cycle_graph(4)
+    >>> G.add_edge(0,3,weight=2)
+    >>> mst = tree.kruskal_mst_edges(G, minimum=False, data=False)
+    >>> sorted(mst)
+    [(0, 1), (0, 3), (1, 2)]
+
+
+    Notes
+    -----
+    Uses Kruskal's algorithm.
+
+    If the graph edges do not have a weight attribute a default weight of 1
+    will be used.
+
+    There may be more than one tree with the same minimum or maximum weight.
+    See :mod:`networkx.tree.recognition` for more detailed definitions.
+
+    Modified code from David Eppstein, April 2006
+    http://www.ics.uci.edu/~eppstein/PADS/
+    """
+    subtrees = UnionFind()
+    edges = sorted(G.edges(data=True), key=lambda t: t[2].get(weight, 1),
+                   reverse=not minimum)
+
+    for u, v, d in edges:
+        if subtrees[u] != subtrees[v]:
+            if data:
+                yield (u, v, d)
+            else:
+                yield (u, v)
+            subtrees.union(u, v)
+
+
+@nx.utils.not_implemented_for('directed')
+def prim_mst_edges(G, minimum, weight='weight', data=True):
+    """Generates the edges of a minimum or maximum spanning tree on `G`.
+
+    Parameters
+    ----------
+    G : undirected graph
+        An undirected graph. If `G` is connected, then the algorithm finds a
+        spanning tree. Otherwise, a spanning forest is found.
+
+    minimum : bool
+       If `True`, find a minimum spanning tree. Otherwise, find a maximum
+       spanning tree.
+
+    weight : str
+       Data key to use for edge weights.
+
+    data : bool
+       If `True`, the edge data is included with each yielded edge.
+
+
+    Returns
+    -------
+    edges : iterator
+        A generator that produces edges in the minimum/maximum spanning tree.
+        The edges are 2-tuples (u, v) or 3-tuples (u, v, d) where u and v
+        define the included edge and d is the edge data.
+
+
+    Examples
+    --------
+    >>> from networkx.algorithms import tree
+
+    Find a minimum spanning tree.
+
+    >>> G = nx.cycle_graph(4)
+    >>> G.add_edge(0, 3, weight=2)
+    >>> mst = tree.prim_mst_edges(G, minimum=True, data=False)
+    >>> sorted(mst)
+    [(0, 1), (1, 2), (2, 3)]
+
+    Find a maximum spanning tree.
+
+    >>> G = nx.cycle_graph(4)
+    >>> G.add_edge(0,3,weight=2)
+    >>> mst = tree.prim_mst_edges(G, minimum=False, data=False)
+    >>> sorted(mst)
+    [(0, 1), (0, 3), (3, 2)]
+
+
+    Notes
+    -----
+    Uses Prim's algorithm.
+
+    If the graph edges do not have a weight attribute a default weight of 1
+    will be used.
+
+    There may be more than one tree with the same minimum or maximum weight.
+    See :mod:`networkx.tree.recognition` for more detailed definitions.
+
+    """
+    push = heappush
+    pop = heappop
+
+    nodes = G.nodes()
```

Comment:
Should we "future-proof" this by using list(G.nodes()).
Or perhaps set(G.nodes()) and adjust the nodes.pop(0) below.